packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts

```diff
@@ -77,22 +76,12 @@ export function useGetEntities(
               header.id,
               header.versionNumber,
             ),
+          options,
         }
       }),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [headerIds],
+    [accessToken, entityHeaders, keyFactory, options],
   )
-  const results = useQueries(queries)
-  const isLoading = results.some(result => result.isLoading)
-  const entities: Entity[] = results
-    .filter(query => query.data !== undefined)
-    .map(query => query.data!)
-  return useMemo(() => {
-    // @ts-ignore
-    if (!isLoading && options?.onSuccess) options.onSuccess(entities)
-    return { isLoading, data: entities }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isLoading, headerIds])
+  return useQueries(queries)
```

Comment:
Just return the result of `useQueries` and let the caller decide what to do with the query results